File: tests/src/org.apache.kafka/kafka-clients/4.2.0/src/test/java/org_apache_kafka/kafka_clients/InternalTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_kafka.kafka_clients;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertSame;

import org.apache.kafka.shaded.com.google.protobuf.Any;
import org.apache.kafka.shaded.com.google.protobuf.ByteString;
import org.apache.kafka.shaded.com.google.protobuf.Internal;
import org.junit.jupiter.api.Test;

public class InternalTest {
    @Test
    void resolvesGeneratedMessageDefaultInstance() {
        Any defaultInstance = Internal.getDefaultInstance(Any.class);

        assertSame(Any.getDefaultInstance(), defaultInstance);
        assertEquals("", defaultInstance.getTypeUrl());
        assertEquals(ByteString.EMPTY, defaultInstance.getValue());
    }
}

File: tests/src/com.typesafe.akka/akka-protobuf-v3_3/2.9.0-M2/src/test/scala/com_typesafe_akka/akka_protobuf_v3_3/InternalTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_typesafe_akka.akka_protobuf_v3_3

import akka.protobufv3.internal.Internal
import akka.protobufv3.internal.{Any => ProtobufAny}
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test

class InternalTest {
  @Test
  def getDefaultInstanceUsesGeneratedMessageAccessor(): Unit = {
    val defaultInstance: ProtobufAny = Internal.getDefaultInstance(classOf[ProtobufAny])

    assertSame(ProtobufAny.getDefaultInstance, defaultInstance)
  }
}
